src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1318,11 +1338,245 @@ class Simplex_tree {
     dimension_ = max_dim - dimension_;
   }
 
+  /**
+    * @brief Adds a vertex or an edge in a flag complex, as well as all
+    * simplices of its star, defined to maintain the property
+    * of the complex to be a flag complex, truncated at dimension dim_max.
+    *
+    * @param[in] u,v              Vertex_handle representing the new edge
+    * @param[in] fil              Filtration value of the edge
+    * @param[in] dim_max          Maximal dimension of the expansion.
+    *                             If set to -1, the expansion goes as far as possible.
+    * @param[out] added_simplices Contains at the end all new
+    *                             simplices induced by the insertion of the edge.
+    *                             The container is not emptied and new simplices are
+    *                             appended at the end.
+    *
+    * @pre `SimplexTreeOptions::link_nodes_by_label` must be true.
+    * @pre When inserting the edge `[u,v]`, the vertices @p u and @p v have to be
+    * already inserted in the simplex tree.
+    *
+    * @warning If the edges and vertices are not inserted in the order of their
+    * filtration values, the method `make_filtration_non_decreasing()` has to be
+    * called at the end of the insertions to restore the intended filtration.
+    * @warning Different from other insertion methods, if a vertex or an edge
+    * was already inserted in the simplex, it will be ignored and the filtration
+    * values will not change.
+    */
+  void insert_edge_as_flag(  Vertex_handle                   u
+                           , Vertex_handle                   v
+                           , Filtration_value                fil
+                           , int                             dim_max
+                           , std::vector<Simplex_handle>&    added_simplices)
```

Comment:
An output iterator or a callback would be a bit more general, but since this is just for use in zigzags, it is ok not to be too general.

Reply:
Should I comment a "TODO" next to it in case the method ends up being used in other cases?

Reply:
no, it is ok. A TODO would be useful for something a bit hidden we might forget. But if in the future we find ourselves forced to do copies to pass this last argument, it will be obvious what the problem is.